tests/cxx/using_unused.cc

```diff
@@ -0,0 +1,21 @@
+//===--- using_unused.cc - test input file for iwyu -----------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+// Tests that when we import a name via a using statement, IWYU tells us we 
+// need to include the original thing we're referencing, despite the fact that
+// it's not actually used.
+
+#include <vector>
+using std::vector;
```

Comment:
Here I think it will be better to use our own class, not STL class. Because for `using` you don't need full use, forward-declare use is sufficient. But we also have a rule not to forward-declare anything from the std namespace [tests/cxx/no_fwd_decl_std.cc](https://github.com/include-what-you-use/include-what-you-use/blob/master/tests/cxx/no_fwd_decl_std.cc). It is better to test these rules separately and not to use `std::vector` in this test case.

And I have discovered a bug that for
```c++
namespace std {
  class vector;
}
using std::vector;
```

IWYU recommends to remove forward declaration that breaks the build. But it's not a regression introduced by your change, master has the same behavior. Need to submit a separate issue for this.